src/xdplwf/bind.c

```diff
@@ -333,6 +321,18 @@ XdpLwfFilterPreDetach(
 
     Filter->PreDetached = TRUE;
 
+    if (Filter->XdpIfInterfaceSetHandle != NULL) {
+        XdpNativeDetachInterface(&Filter->Native);
+        XdpGenericDetachInterface(&Filter->Generic);
+        XdpIfDeleteInterfaceSet(Filter->XdpIfInterfaceSetHandle);
+        XdpNativeWaitForDetachInterfaceComplete(&Filter->Native);
+        XdpGenericWaitForDetachInterfaceComplete(&Filter->Generic);
+    }
+
+    //
+    // TODO: Review: is this necessary if all offload handles are guaranteed to
```

Comment:
It turns out that yes, it is still partially necessary, and the rest was removed. TODO: remove this comment.